meilisearch-core/src/query_builder.rs

```diff
@@ -88,41 +71,74 @@ impl<'c, 'f, 'd> QueryBuilder<'c, 'f, 'd> {
     }
 
     pub fn query(
-        self,
+        mut self,
         reader: &heed::RoTxn<MainT>,
         query: &str,
         range: Range<usize>,
     ) -> MResult<(Vec<Document>, usize)> {
+        let facets_docids = match self.facets.take() {
+            Some(facets) => {
+                let mut ands = Vec::with_capacity(facets.len());
+                let mut ors = Vec::new();
+                for f in facets.deref() {
+                    match f {
+                        Either::Left(keys) => {
+                            ors.reserve(keys.len());
+                            for key in keys {
+                                let docids = self.index.facets.facet_document_ids(reader, &key)?.unwrap_or_default();
+                                ors.push(docids.into_owned());
```

Comment:
I don't understand why we owned it here, it is a Cow that is bounded by the lifetime of the transaction here.
```suggestion
                                ors.push(docids);
```

Reply:
yes right, but then we gottta deref on line 92:
```rust
let sets: Vec<_> = ors.iter().map(Cow::deref).collect();
```